src/main/java/spring/concept/config/AppConfig.java

```diff
@@ -3,25 +3,32 @@
 import java.util.Properties;
 import javax.sql.DataSource;
 import org.apache.commons.dbcp2.BasicDataSource;
-import org.springframework.beans.factory.annotation.Configurable;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
 import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
 import spring.concept.model.User;
 
-@Configurable
+@Configuration
+@PropertySource("classpath:db.properties")
 @ComponentScan(basePackages = {
         "spring.concept.service",
         "spring.concept.dao"
 })
 public class AppConfig {
+    @Autowired
+    private Environment env;
```

Comment:
What is better: setter, constructor, or field injection?

Reply:
Based on question - constructor